### DISCUSSION TITLE
Getting "Invalid date 'APRIL 31'" when specification of time for EXDATE is missing

### DISCUSSION THREAD

**Discussion by bwildenhain** [bitfireAT/ical4android#146] [Q&A] [answered]
Hi,

when trying to sync a Nextcloud calender, which contains the recurring event from
[btest.ics.txt](https://github.com/bitfireAT/davx5-ose/files/14772734/btest.ics.txt)
I get the following error message: `Invalid date 'APRIL 31'`

Although https://icalendar.org/validator.html, GNOME Evolution and Nextcloud Calendar didn't complain and excluded the 31st of may (sic!), the file using only a date where date + time where required for the EXDATE line, so
`EXDATE;VALUE=DATE:20240531`
seems to be wrong wrong, while
`EXDATE;TZID=Europe/Berlin:20240531T114500`
was working. The following debug output was created:

[debug-info.txt](https://github.com/bitfireAT/davx5-ose/files/14773709/debug-info.txt)

I wonder whether it might be a good idea to output a more understandable error message or to handle this more gracefully by accepting the file nevertheless and interpreting it accordingly. I have no experience in kotlin programming, so I didn't look into the actual source code.

**Comment by rfc2822** [comment, marked as answer]
Thanks for the report, should be fixed in the next version (https://github.com/bitfireAT/ical4android/issues/147).
